app/controllers/api/v1/passwords_controller.rb

```diff
@@ -0,0 +1,69 @@
+class API::V1::PasswordsController < API::APIController
+
+    #
+    # Route /api/v1/passwords/reset_password
+    #
+    def reset_password
+
+        login_or_email = params[:login_or_email]
+
+        #
+        # Try to find user by login
+        #
+        user = User.find_by_login(login_or_email)
+    
+        if user.nil?
+            #
+            # Try to find user by email
+            #
+            user = User.find_by_email(login_or_email)
+        end
+      
+        #
+        # Could not find user by login or email?
+        #
+        if user.nil?
+            message = "Cannot find user or email."
+            render status: 403, :json => { :message => message }
+            return
+        end
+    
+        #
+        # Check if user is an SSO user.
+        #
+        if user.is_oauth_user?
+            provider    = user.authentications[0].provider.titleize
+            message     = "This is a #{provider} authenticated account. " <<
+                            "Please use #{provider} to make password changes."
+            render status: 403, :json => { :message => message }
```

Comment:
So in this case you could provide the provider and the reason:
```
{ reason: 'external_auth_user', provider: provider, message: message }
```
That would make it easier for the client to construct the message themselves. if they want.